src/routes/guides/testing-your-app.mdx

```diff
@@ -0,0 +1,357 @@
+```
+
+### Interacting with components
+
+Many components are not static, but instead change on user interactions. These interactions need to be simulated to test the changes. To simulate user interactions, the creators of testing-library have provided the `@testing-library/user-event` library. It takes care of the usual order of events as they would occur in actual user interactions, e.g. that a `click` is accompanied by `mousemove`, `hover`, `keydown`, `focus`, `keyup`, and `keypress`.
+
+If you want to use fake timers with `vi.useFakeTimers()` in your tests, be sure to set it up:
+
+```tsx frame="none"
+import { vi } from "vitest"
+
+const user = userEvent.setup({ advanceTimers: vi.advanceTimersByTime })
+```
+
+The most convenient events are usually `click`, `keyboard` and `pointer` (to simulate touch events). Learn more about them on the [user-event documentation](https://testing-library.com/docs/user-event/intro).
+
+
+### Validating assertions
+
+`vitest` comes with the `expect` function to facilitate assertions that works like:
+
+```tsx frame="none"
+expect(subject)[assertion](value)
+```
+
+The command supports assertions like `toBe` (reference comparison) and `toEqual` (value comparison) out of the box. For testing inside the DOM, the package `@testing-library/jest-dom` augments it with some helpful additional assertions:
+
+- `.toBeInTheDocument()` - checks if the element actually exists in the DOM
+- `.toBeVisible()` - checks if there is no reason the element should be hidden
+- `.toHaveTextContent(content)` - checks if the text content matches
+- `.toHaveFocus()` - checks if this is the currently focused element
+- `.toHaveAccessibleDescription(description)` - checks accessible description
+- and a [lot more](https://github.com/testing-library/jest-dom?tab=readme-ov-file#custom-matchers).
+
+
+## Directive testing
+
+[Directives](/reference/jsx-attributes/use) are reusable behavior for elements. As such, they get a reference to the element and the property value given to them as an accessor. To make testing them more concise, our `@solidjs/testing-library` has a renderDirective function:
+
+
+```ts frame="none"
+const renderResult = renderDirective(directive, {
+  initialValue, // value initially added to the argument signal
+  targetElement, // opt. node name or element used as target for the directive
+  ...renderOptions, // see render options
+})
+const {
+  arg, // getter for the directive's argument
+  setArg, // setter for the directive's argument
+  ...renderResults, // see render results
+} = renderResult
+```
+
+The `container` from the `...renderResults` will contain the targetElement (default is a `<div>`). That and the ability to manipulate the `arg` signal are helpful in testing directives.
+
+If for example there is a [directive that handles the Fullscreen API](https://github.com/solidjs-community/solid-primitives/blob/main/packages/fullscreen/src/index.ts), it could be tested like this:
```

Comment:
I am not sure about the guidelines regarding this, and perhaps @LadyBluenotes could give me quick help. But does it make sense to link people away for something that will be affected from the next test? Perhaps we should find a way of having this snippet here

Reply:
I can include the code (or a reduced version) as an example.